3rdParty/iresearch/core/index/index_writer.cpp

```diff
@@ -878,7 +888,19 @@ index_writer::flush_context_ptr index_writer::documents_context::update_segment(
   return ctx;
 }
 
-void index_writer::flush_context::emplace(active_segment_context&& segment) {
+void index_writer::flush_context::AddToPending(
+    active_segment_context& segment) {
+  if (segment.flush_ctx_) {
+    return;
+  }
+  auto lock = make_lock_guard(mutex_);
+  segment.flush_ctx_ = this;
+  segment.pending_segment_context_offset_ = pending_segment_contexts_.size();
+  pending_segment_contexts_.emplace_back(segment.ctx_,
```

Comment:
If the `emplace_back` operation throws, the segment will already be modified. Do the modifications need to be rolled back then?